src/services/bitcoin.js

```diff
@@ -0,0 +1,161 @@
+
+import { DEFAULT_BTC_NETWORK } from 'constants/bitcoinConstants';
+import type {
+  BitcoinUtxo,
+  BitcoinTransactionTarget,
+  BitcoinTransactionPlan,
+} from 'models/Bitcoin';
+import { SPEED_TYPES } from 'constants/assetsConstants';
+import {
+  getAddressUtxosFromNode,
+  sendRawTransactionToNode,
+} from 'services/insight';
+
+const bip39 = require('bip39');
+const bip32 = require('bip32');
+const coinselect = require('coinselect');
+
+const NETWORK = networks[DEFAULT_BTC_NETWORK];
+
+const feeRateFromSpeed = (speed: string): number => {
+  switch (speed) {
+    // TODO: define rates
+    case SPEED_TYPES.SLOW: return 50;
+    case SPEED_TYPES.NORMAL: return 50;
+    case SPEED_TYPES.FAST: return 50;
+    default: return 50;
+  }
+};
+
+export const collectOutputs = (
+  targets: BitcoinTransactionTarget[],
+  speed: string,
+  unspent: BitcoinUtxo[],
+  changeAddress: (value: number) => string,
+): BitcoinTransactionPlan => {
+  const feeRate = feeRateFromSpeed(speed);
+
+  const utxos = unspent.map(utxo => ({ ...utxo, value: utxo.satoshis }));
+  const {
+    inputs,
+    outputs,
+    fee,
+  } = coinselect(utxos, targets, feeRate);
+
+  const isValid = !!(inputs && outputs);
+
+  const planOutputs = isValid ? outputs.map(out => {
+    if (out.address) {
+      return { ...out, isChange: false };
+    }
+    const address = changeAddress(out.value);
+
+    return { ...out, address, isChange: true };
+  }) : [];
+  const planInputs = isValid ? inputs : [];
+
+  return {
+    fee,
+    isValid,
+    inputs: planInputs,
+    outputs: planOutputs,
+  };
+};
+
+export const sendRawTransaction = async (rawTx: string): Promise<string> => {
+  return sendRawTransactionToNode(rawTx)
+    .then(response => response.json())
+    .then(({ txid }) => txid)
+    .catch(() => null);
+};
+
+export const transactionFromPlan = (
+  plan: BitcoinTransactionPlan,
+  inputSigner: (address: string) => ECPair,
+  networkName?: string,
+): string => {
+  const network = networkName ? networks[networkName] : NETWORK;
+
+  const txb = new TransactionBuilder(network);
+  txb.setVersion(1);
+
+  plan.inputs.forEach(utxo => {
+    txb.addInput(
+      utxo.txid,
+      utxo.vout,
+    );
+  });
+  plan.outputs.forEach(out => {
+    txb.addOutput(out.address, out.value);
+  });
+
+  let utxoIndex = 0;
+  plan.inputs.forEach(({ address }) => {
+    const keyPair = inputSigner(address);
+
+    txb.sign({
+      keyPair,
+      prevOutScriptType: 'p2pkh',
+      vin: utxoIndex++,
+    });
+  });
+
+  return txb.build().toHex();
+};
+
+export const rootFromMnemonic = async (mnemonic: string, networkName?: string): ECPair => {
+  const network = networkName ? networks[networkName] : NETWORK;
+  const seed = await bip39.mnemonicToSeed(mnemonic);
+
+  return bip32.fromSeed(seed, network);
+};
+
+export const keyPairAddress = (keyPair: ECPair): string => {
+  const { address } = payments.p2pkh({
+    pubkey: keyPair.publicKey,
+    network: keyPair.network,
+  });
```

Comment:
You sure this will always return object that can be deconstructed?

Reply:
Just checked and I think not – https://github.com/bitcoinjs/bitcoinjs-lib/blob/19a126c205dee75abbbd9ab185b11eca3996d587/src/payments/p2pkh.js.

I think adding either `lodash.get` or `|| {}` would  make sense

Reply:
This is always gonna return an object or throw an error. Updated accordingly.